File: test/fakers/banner_faker.dart
import 'package:faker/faker.dart';
import 'package:sertton/core/marketing/dtos/banner_dto.dart';

class BannerFaker {
  static final _faker = Faker();

  static BannerDto fakeDto({String? id, String? imageUrl}) {
    return BannerDto(
      id: id ?? _faker.guid.guid(),
      imageUrl: imageUrl ?? _faker.image.loremPicsum(),
    );
  }

  static List<BannerDto> fakeManyDto({
    int count = 10,
    String? id,
    String? imageUrl,
  }) {
    return List.generate(count, (index) => fakeDto(id: id, imageUrl: imageUrl));
  }
}

File: src/core/marketing/entities/fakers/banner-faker.ts
import { faker } from "@faker-js/faker"

import type { Banner } from "../banner"

export class BannerFaker {
  static fake(banner?: Partial<Banner>): Banner {
    return { id: faker.string.uuid(), imageUrl: faker.image.url(), ...banner }
  }

  static fakeMany(count = 10, banner?: Partial<Banner>): Banner[] {
    return Array.from({ length: count }, () => BannerFaker.fake(banner))
  }
}
